foreshadow/intents/registry.py

```diff
@@ -34,6 +36,78 @@ def validate_input(clsname):
         raise ValueError("Input must be either a string or a list of strings")
 
 
+def _process_templates(cls_target):
+    def _resolve_template(template):
+        if not all(len(s) == 3 for s in template) or not template:
+            raise ValueError("Malformed template")
+        if not all(
+            (
+                isinstance(s[1], type)
+                and issubclass(s[1], (BaseEstimator, TransformerMixin))
+            )
+            or (
+                isinstance(s[1], tuple)
+                and len(s[1]) == 2
+                and isinstance(s[1][0], type)
```

Comment:
Can we make the pipeline template a named tuple? Using index to access a list assume everyone remember the meaning of every position. It might be fine now, but consider we need to maintain this code a year from now?

Reply:
This is a good suggestion, I have implemented it in the updated pull request that also merges some changes that Alexander had made.